protocols/1617/csv_cherrypick.ot2.py

```diff
@@ -0,0 +1,168 @@
+    transfer_info.pop(0)
+    source_plate_slots = []
+    source_wells = []
+    target_plate_slots = []
+    target_wells = []
+    volumes = []
+    mix_vols = []
+    mix_cycles = []
+
+    # parse for transfer information
+    for line in transfer_info:
+        source_plate_slots.append(line[0].strip())
+        source_wells.append(line[1].strip())
+        volumes.append(float(line[2].strip()))
+        target_plate_slots.append(line[3].strip())
+        target_wells.append(line[4].strip())
+        mix_vols.append(float(line[5].strip()))
+        mix_cycles.append(int(line[6].strip()))
+
+    unique_source_slots = {}
+    source_plate_inds = 0
+    for slot in source_plate_slots:
+        if slot not in unique_source_slots:
+            unique_source_slots.update({slot: source_plate_inds})
+            source_plate_inds += 1
+
+    unique_target_slots = {}
+    target_plate_inds = 0
+    for slot in target_plate_slots:
+        if slot not in unique_target_slots:
+            unique_target_slots.update({slot: target_plate_inds})
+            target_plate_inds += 1
+
+    # load proper labware
+    source_plates = [labware.load(source_name, key, 'source ' + key)
+                     for key in unique_source_slots]
+    destination_plates = [labware.load(dest_name, key, 'destination ' + key)
+                          for key in unique_target_slots]
+
+    # pipette and tiprack setup depending on pipette selection
+    if pipette_selection.split(' ')[0] == 'P10':
+        tips10 = [labware.load('tiprack-10ul', slot) for slot in ['10', '11']]
+        tips300 = [labware.load('opentrons-tiprack-300ul', slot)
+                   for slot in ['7', '8', '9']]
+        tips10_max = 96*2
+        tips300_max = 96*3
+        all_tips_10 = [well for rack in tips10 for well in rack.wells()]
+        p10 = instruments.P10_Single(mount='right')
+        if pipette_selection.split(' ')[2] == 'P50':
+            p50 = instruments.P50_Single(mount='left')
+        else:
+            p300 = instruments.P300_Single(mount='left')
+        tip10_count = 0
+    else:
+        tips300 = [labware.load('opentrons-tiprack-300ul', slot)
+                   for slot in ['7', '8', '9', '10', '11']]
+        tips300_max = 96*5
+        p50 = instruments.P50_Single(mount='right')
+        p300 = instruments.P300_Single(mount='left')
+    all_tips_300 = [well for rack in tips300 for well in rack.wells()]
+    tip300_count = 0
+
+    # perform transfers
+    for s_slot, s_well, t_slot, t_well, vol, mix_vol, mix_n in zip(
+            source_plate_slots,
+            source_wells,
+            target_plate_slots,
+            target_wells,
+            volumes,
+            mix_vols,
+            mix_cycles):
+
+        if pipette_selection.split(' ')[0] == 'P10':
+            if tip10_count == tips10_max:
+                robot.pause('Replace tipracks before resuming.')
+                p10.reset()
```

Comment:
Would you also need to reset `tip10_count` and `tip300_count` to 0 after the tipracks are replaced?